### DISCUSSION TITLE
FFT looks odd or wrong somehow, a code issue or just the ESP32s poor ADC ?

### DISCUSSION THREAD

**Discussion by jonshouse1** [PlummersSoftwareLLC/NightDriverStrip#104] [Show and tell]
Re: FFT looks wrong somehow from youtube....

When I did my audio projects many years back I used a couple of FFT routines that I modified a bit. See "player_fft.c" and "player_integer_fft.c" from "src_v0.61.tar.gz" here:

https://sourceforge.net/projects/centroplayer/files/

Results are here:
https://www.youtube.com/watch?v=zHSGtInrsDY&t=62s

Not a fair comparison with the ESP32 though as this was CD quality audio on a fast ARM CPU.  Still might be worth a quick look  at in case it is better than the aurdino ide default fft.

**Comment by rbergen** [comment]
I'm converting this to a show and tell discussion. Feel free to change the title of the discussion to something that more closely matches its content. :)

**Reply by jonshouse1** [reply]
Thanks, hope the new title is ok.

**Reply by rbergen** [reply]
Sure looks ok to me. If you would like me to change the category of this conversation from "Show and tell" to another one, then feel free to let me know.

**Reply by jonshouse1** [reply]
The category is up to you! 
My point was more that the FFT code from my project might be better than the default lib (I can't tell). It should just drop in and build. My project used two FFT routines, an integer only one (as the original RIO/Empeg hardware was an Arm with no FPU) and a hard float version for platforms with reasonable speed floating point.
